### PR TITLE
Update for pending strictness change in layering check

### PR DESCRIPTION
**Pull request by jpienaar**
Post https://reviews.llvm.org/D132779 the layering check will be
stricter.

### CODE REVIEW

File: runtime/src/iree/modules/hal/utils/BUILD [hunk 1]
```diff
@@ -22,5 +22,6 @@ iree_runtime_cc_library(
         "//runtime/src/iree/hal",
         "//runtime/src/iree/modules/hal:types",
         "//runtime/src/iree/vm",
+        "//runtime/src/iree/vm:impl",
```
Comment (benvanik) [inline runtime/src/iree/modules/hal/utils/BUILD:25]: impl shouldn't be used outside of vm/... - what header is getting pulled in here?

Comment (benvanik) [reply runtime/src/iree/modules/hal/utils/BUILD:25]: ah - you want to drop the `#include "iree/vm/shims.h"` from buffer_diagnostics.c - iree/vm/api.h includes that

Comment (jpienaar) [reply runtime/src/iree/modules/hal/utils/BUILD:25]: Yeah that one looked weird. 

iree/runtime/src/iree/vm/shims.h is flagged as being in impl

Comment (jpienaar) [reply runtime/src/iree/modules/hal/utils/BUILD:25]: (sorry github didn't show me your message before I responded)

**Review by benvanik — APPROVED** [review]

**Review by benvanik — CHANGES_REQUESTED** [review]
actually lets see if we can drop impl from bytecode_module too - maybe bytecode_disasm.h changing from including stack.h to api.h?

**Comment by benvanik** [comment]
I'm excited to have the layering checks enabled - these kind of things bitrot fast without it - thanks for doing this!

**Review by benvanik — APPROVED** [review]

**Pull request merged**